backend/src/StdLibExecution/Libs/HttpClient.fs

```diff
@@ -4,18 +4,233 @@
 open Prelude
+open LibExecution
 open LibExecution.RuntimeTypes
 
+open VendoredTablecloth
+
 open LibExecution.StdLib.Shortcuts
 
 
-let types : List<BuiltInType> = []
+module HttpClient =
+  type Method = HttpMethod
+
+  module Headers =
+    type Header = string * string
+    type T = List<Header>
+
+  type Body = byte array
+
+  type HttpRequest =
+    { url : string
+      method : Method
+      headers : Headers.T
+      body : Body }
+
+  type HttpResult = { statusCode : int; headers : Headers.T; body : Body }
+
+  // forked from Elm's HttpError type
+  // https://package.elm-lang.org/packages/elm/http/latest/Http#Error
+  type HttpRequestError =
+    | BadUrl of details : string
+    | Timeout
+    | NetworkError
+    | Other of details : string
+
+  type HttpRequestResult = Result<HttpResult, HttpRequestError>
+
+  // There has been quite a history of .NET's HttpClient having problems,
+  // including socket exhaustion and DNS results not expiring.
+  // The history is outlined well here:
+  // https://www.stevejgordon.co.uk/httpclient-connection-pooling-in-dotnet-core
+  //
+  // As of .NET 6 it seems we no longer need to worry about either socket
+  // exhaustion or DNS issues. It appears that we can use either multiple HTTP
+  // clients or just one, we use just one for efficiency.
+  // See https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-7.0#alternatives-to-ihttpclientfactory
+  //
+  // Note that the number of sockets was verified manually, with:
+  // `sudo netstat -apn | grep _WAIT`
+  // TODO: I don't see where "the number of sockets" is actually configured?
+  // let private socketHandler : HttpMessageHandler =
+  //   new SocketsHttpHandler(
+  //     // Avoid DNS problems
+  //     PooledConnectionIdleTimeout = System.TimeSpan.FromMinutes 5.0,
+  //     PooledConnectionLifetime = System.TimeSpan.FromMinutes 10.0,
+
+  //     // HttpClientTODO avail functions to compress/decompress with common
+  //     // compression algorithms (gzip, brottli, deflate)
+  //     //
+  //     // HttpClientTODO consider: is there any reason to think that ASP.NET
+  //     // does something fancy such that automatic .net httpclient -level
+  //     // decompression would be notably more efficient than doing so 'manually'
+  //     // via some function? There will certainly be more bytes passed around -
+  //     // probably not a big deal?
+  //     AutomaticDecompression = System.Net.DecompressionMethods.None,
+
+  //     // HttpClientTODO avail function that handles redirect behaviour
+  //     AllowAutoRedirect = false,
+
+  //     // UseProxy = true,
+  //     // Proxy = System.Net.WebProxy(Config.httpclientProxyUrl, false),
+
+  //     // Don't add a RequestId header for opentelemetry
+  //     ActivityHeadersPropagator = null,
+
+  //     // Users share the HttpClient, don't let them share cookies!
+  //     UseCookies = false
+  //   )
+
+  let private httpClient : HttpClient =
+    new HttpClient(
```

Comment:
second, it turned out that by providing _any_ sockets handler to the `new HttpClient()`, it overrode the Blazor functionality of using the browser's `fetch` for the http client, so we reduced it to a more boring `new HttpClient()`